k8s/apps/deployments.go

```diff
@@ -293,3 +296,35 @@ func (c *Client) GetDeploymentsUsingStorageClass(scName string) ([]appsv1.Deploy
 
 	return retList, nil
 }
+
+// DeletePodsByDeployment deletes pods for the given deployment name and namespace
+func (c *Client) DeletePodsByDeployment(name, namespace string, timeout time.Duration) error {
+	deployment, err := c.GetDeployment(name, namespace)
+	if err != nil {
+		return err
+	}
+
+	pods, err := c.GetDeploymentPods(deployment)
+	if err != nil {
+		return err
+	}
+
+	var podsNamesToDelete []string
+	var podsToDelete []corev1.Pod
+	for _, pod := range pods {
+		podsNamesToDelete = append(podsNamesToDelete, pod.Name)
+		podsToDelete = append(podsToDelete, pod)
+	}
+
+	logrus.Debugf("Deleting pods: %s", podsNamesToDelete)
+	if err := common.DeletePods(c.core, pods, false); err != nil {
+		return err
+	}
+
+	if err := common.WaitForPodsToBeDeleted(c.core, podsToDelete, timeout); err != nil {
+		return fmt.Errorf("Failed to wait for pods to be deleted: %s, Err: %v", podsNamesToDelete, err)
+	}
+
+	logrus.Debugf("Pods for deployment %s have been successfully deleted", name)
```

Comment:
Let's not log from here for consistency. Also, we don't know if every caller is using logrus or not. The caller can always print these two logs before and after their call.

Reply:
Done, remove logrus.Debugf()